### PR TITLE
Re-generate API token on re-provisioning [EN-1039]

### PR DESCRIPTION
**Pull request by gaborszakacs**
In order to seamlessly rotate API tokens generated by add-ons
and stored in Website, we need to support re-provisioning
of already existing apps - we need to regenerate
and send a new token in this case.

Manually tested in development environment:
- app provisioning (with authorization switched off and Website API calls disabled): returned with api token and persisted in DB
- same app provisioned again: returned with new api token and persisted it in DB, other data hasn't changed

### CODE REVIEW

File: services/provision.go [hunk 1]
```diff
@@ -67,13 +67,24 @@ func ProvisionHandler(env *env.AppEnv, w http.ResponseWriter, r *http.Request) e
...
+		var err error
+		app.APIToken = crypto.SecureRandomHash(50)
+		verrs, err = env.AppService.Update(app, []string{"APIToken"})
+		if len(verrs) > 0 {
+			return httpresponse.RespondWithUnprocessableEntity(w, verrs)
+		}
+		if err != nil {
+			return errors.Wrap(err, "SQL Error")
+		}
 	case err != nil:
```
Comment (miklosboros) [inline services/provision.go:81]: This case should be "default" as the condition at this point is always true.

File: services/provision.go [hunk 2]
```diff
@@ -67,13 +67,24 @@ func ProvisionHandler(env *env.AppEnv, w http.ResponseWriter, r *http.Request) e
 		if err != nil {
 			return errors.WithStack(err)
 		}
+	case err == nil:
+		var verrs []error
+		var err error
+		app.APIToken = crypto.SecureRandomHash(50)
```
Comment (miklosboros) [inline services/provision.go:73]: Why 50?

Comment (gaborszakacs) [reply services/provision.go:73]: Following the existing convention:
https://github.com/bitrise-io/addons-ship-backend/blob/b5eaa52d34a0bae8cf795514a81dc76e3ba6a353/services/provision.go#L57

File: services/provision.go (hunk 1 above)
Comment (gaborszakacs) [reply services/provision.go:81]: I see no harm in emphasizing that's an error case here. What's the downside?

Comment (miklosboros) [reply services/provision.go:81]: It can be hard to follow at first glance that there are no more possible cases, but not really an issue.

File: services/provision.go (hunk 2 above)
Comment (miklosboros) [reply services/provision.go:73]: Okay, just wanted to ask from where this number originates.
